File: src/java/_08final/mvc/model/prime/PolarPoint.java
package _08final.mvc.model.prime;

//this record is used in conjunction with Point[] for rendering vector graphics.
//r corresponds to the hypotenuse in cartesian, a number between 0.0 and 1.0.
//theta is degrees expressed in radians, a number between 0.0 and 6.283 (2 * Pi).
public record PolarPoint(double r, double theta) {
}

File: src/python/_08final/mvc/model/prime/PolarPoint.py
from dataclasses import dataclass


# this record (immutable dataclass) is used in conjunction with Point to render vector graphics.
# r corresponds to the hypotenuse in cartesian (0.0-1.0); theta is in radians (0.0-2*Pi).
@dataclass(frozen=True)
class PolarPoint:
    r: float
    theta: float
